azuredevops/internal/service/audit/common.go

```diff
@@ -0,0 +1,186 @@
+func doBaseExpansion(d *schema.ResourceData) (*audit.AuditStream, *int) {
+	// an "error" is OK here as it is expected in the case that the ID is not set in the resource data
+	var auditStreamId *int
+	parsedId, err := strconv.Atoi(d.Id())
+	if err == nil {
+		auditStreamId = &parsedId
+	}
+
+	daysToBackfill := converter.Int(d.Get("days_to_backfill").(int))
+	auditStream := &audit.AuditStream{
+		Id: auditStreamId,
+	}
+
+	return auditStream, daysToBackfill
+}
+
+// doBaseFlattening performs the flattening for the 'base' attributes that are defined in the schema, above
+func doBaseFlattening(d *schema.ResourceData, auditStream *audit.AuditStream, daysToBackfill *int) {
+	d.SetId(strconv.Itoa(*auditStream.Id))
+	d.Set("days_to_backfill", daysToBackfill)
+}
+
+func genAuditStreamCreateFunc(flatFunc flatFunc, expandFunc expandFunc) func(d *schema.ResourceData, m interface{}) error {
+	return func(d *schema.ResourceData, m interface{}) error {
+		clients := m.(*client.AggregatedClient)
+		auditStream, daysToBackfill, err := expandFunc(d)
+		if err != nil {
+			return fmt.Errorf(errMsgTfConfigRead, err)
+		}
+
+		createdAuditStream, err := createAuditStream(clients, auditStream, daysToBackfill)
+		if err != nil {
+			return fmt.Errorf("Error creating audit stream in Azure DevOps: %+v", err)
+		}
+
+		d.SetId(strconv.Itoa(*createdAuditStream.Id))
+		return genAuditStreamReadFunc(flatFunc)(d, m)
+	}
+}
+
+func genAuditStreamReadFunc(flatFunc flatFunc) func(d *schema.ResourceData, m interface{}) error {
+	return func(d *schema.ResourceData, m interface{}) error {
+		clients := m.(*client.AggregatedClient)
+		streamId, err := strconv.Atoi(d.Id())
+		if err != nil {
+			return fmt.Errorf("Error parsing the audit stream ID from the Terraform resource data: %v", err)
+		}
+
+		daysToBackfill := d.Get("days_to_backfill").(int)
+
+		auditStream, err := readAuditStream(clients, streamId)
+		if err != nil {
+			if utils.ResponseWasNotFound(err) {
+				d.SetId("")
+				return nil
+			}
+			return fmt.Errorf("Error looking up audit stream with ID %d. Error: %v", streamId, err)
+		}
+
+		if auditStream.Id == nil {
+			// e.g. audit stream has been deleted separately without TF
+			d.SetId("")
+		} else {
+			flatFunc(d, auditStream, &daysToBackfill)
+		}
+		return nil
+	}
+}
+
+func genAuditStreamUpdateFunc(flatFunc flatFunc, expandFunc expandFunc) schema.UpdateFunc {
+	return func(d *schema.ResourceData, m interface{}) error {
+		clients := m.(*client.AggregatedClient)
+		auditStream, daysToBackfill, err := expandFunc(d)
+		if err != nil {
+			return fmt.Errorf(errMsgTfConfigRead, err)
+		}
+
+		updatedAuditStream, err := updateAuditStream(clients, auditStream)
+		if err != nil {
+			return fmt.Errorf("Error updating audit stream in Azure DevOps: %+v", err)
+		}
+
+		flatFunc(d, updatedAuditStream, daysToBackfill)
+		return genAuditStreamReadFunc(flatFunc)(d, m)
+	}
+}
+
+func genAuditStreamDeleteFunc(expandFunc expandFunc) schema.DeleteFunc {
```

Comment:
`expandFunc` is not used within this function , can be removed

Reply:
removed in latest push